include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4630,6 +4630,22 @@ struct OrtApi {
    * \snippet{doc} snippets.dox OrtStatus Return Value
    */
   ORT_API2_STATUS(KernelInfoGetAllocator, _In_ const OrtKernelInfo* info, _In_ OrtMemType mem_type, _Outptr_ OrtAllocator** out);
+
+  /** \brief This function allows to specify which rules and transformers can be disabled for your model
+   *  to save on model loading time. Pre-optimizing the model is still the recommended approach.
+   *  When loading a pre-optimized model one can tune or disable optimization level using
+   *  SetSessionGraphOptimizationLevel. This function replaces the content of any previously specified
+   *  names within the SessionOptions instance.
+   *
+   * \param[in, out] sess_options instance of OrtSessionOptions
+   * \param[in] disabled_rules_and_transformers_names array of utf-8 encoded zero terminated strings with the
+                names of the rules and transformers to be disabled.
+   * \param[in] numes_num length of disabled_rules_and_transformers_names array
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   */
+  ORT_API2_STATUS(DisableRulesAndOptimizers, _Inout_ OrtSessionOptions* sess_options,
```

Comment:
1. Can we add this to `onnxruntime_session_options_config_keys.h` instead? This way we don't have to write a binding for every single language; all bindings will get it by default. Secondly, since the C API already uses a name called `GraphOptimization`, we can name this config to something like `disable_graph_optimizers` with value being a comma separated list of names.
2. Moreover, it's unfortunate how the Python API has been exposed. If we put this in the config_keys file, we can deprecate the existing Python API.
3. How does a user know what names are available to disable?